rocksdb_admin/admin_handler.cpp

```diff
@@ -415,8 +451,48 @@ AdminHandler::AdminHandler(
   CHECK(FLAGS_max_s3_sst_loading_concurrency > 0)
     << "Invalid FLAGS_max_s3_sst_loading_concurrency: "
     << FLAGS_max_s3_sst_loading_concurrency;
+
+  if (FLAGS_enable_async_delete_dbs) {
+    static const std::string db_tmp_path = FLAGS_rocksdb_dir + "db_tmp/";
+    if (!boost::filesystem::exists(db_tmp_path)) {
+      boost::system::error_code create_err;
+      boost::filesystem::create_directories(db_tmp_path, create_err);
+      if (create_err) {
+        LOG(ERROR) << "Failed to create dir: " << db_tmp_path << create_err.message();
+        return;
+      }
+    }
+
+    db_deletion_evb_ = std::make_unique<folly::EventBase>();
+    db_deletion_thread_ = std::make_unique<std::thread>([evb = db_deletion_evb_.get()] {
+      if (!folly::setThreadName("DBDeleter")) {
+        LOG(ERROR) << "Failed to set thread name for DB deletion thread";
+      }
+
+      LOG(INFO) << "Starting DB deletion thread ...";
+      evb->loopForever();
+      LOG(INFO) << "Stopping DB deletioin thread ...";
+    });
+
+    scheduleDeleteDBs();
+  }
```

Comment:
why do we need an event base? can we just let the thread do the deletion work synchronously?

Reply:
the point is when we just use a thread with x min sleep, then for deployment, we may have to wait for x mins to stop and join the thread. use evb->terminateLoopSoon() could help us to avoid it